x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -153,17 +137,13 @@ public Realms(Settings settings, Environment env, Map<String, Realm.Factory> fac
         if (licenseState.isAuthAllowed() == false) {
             return Collections.emptyList();
         }
```

Comment:
While this may never happen in practice, we need to have a consistent view of the license state for correctness here.

```suggestion
        final XPackLicenseState licenseStateSnapshot = copyCurrentLicenseState();
        if (licenseStateSnapshot.isAuthAllowed() == false) {
            return Collections.emptyList();
        }
```